components/relevancy/src/db.rs

```diff
@@ -11,47 +11,57 @@ use crate::{
 use parking_lot::Mutex;
 use rusqlite::{Connection, OpenFlags};
 use sql_support::{open_database::open_database_with_flags, ConnExt};
-use std::path::Path;
+use std::path::{Path, PathBuf};
 
 /// A thread-safe wrapper around an SQLite connection to the Relevancy database
 pub struct RelevancyDb {
-    pub conn: Mutex<Connection>,
+    inner: Mutex<RelevancyDbInner>,
+}
+
+struct RelevancyDbInner {
+    db_path: PathBuf,
+    reader: Option<Connection>,
+    writer: Option<Connection>,
 }
 
 impl RelevancyDb {
-    pub fn open(path: impl AsRef<Path>) -> Result<Self> {
-        let conn = open_database_with_flags(
-            path,
-            OpenFlags::SQLITE_OPEN_URI
-                | OpenFlags::SQLITE_OPEN_NO_MUTEX
-                | OpenFlags::SQLITE_OPEN_CREATE
-                | OpenFlags::SQLITE_OPEN_READ_WRITE,
-            &RelevancyConnectionInitializer,
-        )?;
+    pub fn new(path: impl AsRef<Path>) -> Result<Self> {
         Ok(Self {
-            conn: Mutex::new(conn),
+            inner: Mutex::new(RelevancyDbInner {
+                db_path: path.as_ref().to_owned(),
+                reader: None,
+                writer: None,
+            }),
         })
     }
 
+    pub fn close(&self) {
+        let mut inner = self.inner.lock();
+        inner.writer = None;
```

Comment:
Just to confirm, we don't need to call `conn.close()`, right? As it will be done automatically by `drop()`;